File: common/src/main/java/cc/turtl/turtlshell/command/InfoCommand.java
package cc.turtl.turtlshell.command;

import cc.turtl.turtlshell.TurtlShellConstants;
import cc.turtl.turtlshell.util.MessageUtils;
import com.mojang.brigadier.Command;
import com.mojang.brigadier.builder.LiteralArgumentBuilder;
import com.mojang.brigadier.context.CommandContext;
import net.minecraft.client.Minecraft;
import net.minecraft.client.player.LocalPlayer;

public class InfoCommand implements TurtlShellCommand {

    @Override
    public String getName() {
        return "info";
    }

    @Override
    public String getDescription() {
        return "Display mod info";
    }

    @Override
    public <S> LiteralArgumentBuilder<S> build() {
        return LiteralArgumentBuilder.<S>literal(getName())
                .executes(this::execute);
    }

    private <S> int execute(CommandContext<S> context) {
        LocalPlayer player = Minecraft.getInstance().player;
        if (player == null) return 0;

        MessageUtils.sendEmptyLine(player);
        MessageUtils.sendSuccess(player, TurtlShellConstants.MOD_DISPLAY_NAME + " Info");
        MessageUtils.sendLabeled(player, "  Version", TurtlShellConstants.VERSION);
        MessageUtils.sendLabeled(player, "  Author", TurtlShellConstants.AUTHOR);
        return Command.SINGLE_SUCCESS;
    }
}

File: common/src/main/kotlin/cc/turtl/turtlshell/command/InfoCommand.kt
package cc.turtl.turtlshell.command

import cc.turtl.turtlshell.TurtlShellConstants
import cc.turtl.turtlshell.api.TurtlShellCommand
import cc.turtl.turtlshell.util.format.ColorLib
import cc.turtl.turtlshell.util.format.component.INDENT
import cc.turtl.turtlshell.util.format.component.MOD_PREFIX
import cc.turtl.turtlshell.util.format.component.NEW_LINE
import cc.turtl.turtlshell.util.format.component.componentOf
import com.mojang.brigadier.Command
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import net.minecraft.commands.CommandSourceStack
import net.minecraft.network.chat.Component

object InfoCommand : TurtlShellCommand {
    override val name = "info"
    override val description = "Display mod info"

    override fun build(): LiteralArgumentBuilder<CommandSourceStack> =
        LiteralArgumentBuilder.literal<CommandSourceStack>(name).executes { context ->

            val message = Component.empty()
                .append(NEW_LINE)
                .append(MOD_PREFIX)
                .append(componentOf("${TurtlShellConstants.MOD_DISPLAY_NAME} Info").withColor(ColorLib.MINT.rgb))

            message.append(NEW_LINE)
                .append(MOD_PREFIX)
                .append(INDENT)
                .append("Version: ${TurtlShellConstants.VERSION}")

            message.append(NEW_LINE)
                .append(MOD_PREFIX)
                .append(INDENT)
                .append("Author: ${TurtlShellConstants.AUTHOR}")

            context.source.sendSystemMessage(message)

            Command.SINGLE_SUCCESS
        }
}